lib/jsonSchema.ts

```diff
@@ -1,127 +1,159 @@
-        "extends",
-    ];
-    for (const field in unsupportedFields) {
-        if (!(js[field] === undefined || js[field] === false)) {
-            return false;
-        }
+  const unsupportedFields = [
+    "$ref",
+    "multipleOf",
+    "allOf",
+    "anyOf",
+    "oneOf",
+    "not",
+    "additionalItems",
+    "enum",
+    "extends",
+  ]
+  for (const field in unsupportedFields) {
+    if (!(js[field] === undefined || js[field] === false)) {
+      return false
     }
-    return true;
+  }
+  return true
 }
 
 /**
  * Error to capture all the unsupported edge cases
  */
 export class UnsupportedJsonSchemaError extends Error {
-    constructor(msg: string) {
-        const message = `Unsupported JSON schema: ${msg}`;
-        super(message);
-        this.name = 'UnsupportedJsonSchemaError';
-    }
+  constructor(msg: string) {
+    const message = `Unsupported JSON schema: ${msg}`
+    super(message)
+    this.name = "UnsupportedJsonSchemaError"
+  }
 }
 
 /**
  * Json Schema has required at the top level instead of field level
  */
-const isJsonSchemaRequired = (jsonSchema: SupportedJSONSchema4) => (field: string): boolean => {
+const isJsonSchemaRequired =
+  (jsonSchema: SupportedJSONSchema4) =>
+  (field: string): boolean => {
     switch (jsonSchema.required) {
-        case true: return true;
-        case undefined:
-        case false:
-            return false;
+      case true:
+        return true
+      case undefined:
+      case false:
+        return false
     }
 
-    return jsonSchema.required.includes(field);
-}
+    return jsonSchema.required.includes(field)
+  }
 
 /**
  * Converts the Array field type into the correct Field Definition
  */
-const fromJsonSchemaArray = (fieldValue: SupportedJSONSchema4, optionalFieldList: boolean): FieldDefinition => {
-    if (!fieldValue.items || !fieldValue.items.type) {
-        throw new UnsupportedJsonSchemaError("Array field with no values found.");
-    }
+const fromJsonSchemaArray = (
+  fieldValue: SupportedJSONSchema4,
+  optionalFieldList: boolean
+): FieldDefinition => {
+  if (!fieldValue.items || !fieldValue.items.type) {
+    throw new UnsupportedJsonSchemaError("Array field with no values found.")
+  }
 
-    switch (fieldValue.items.type) {
-        case 'string':
-            return fields.createListField('UTF8', optionalFieldList);
-        case 'integer':
-        case 'number':
-            return fields.createListField('INT64', optionalFieldList);
-        case 'boolean':
-            return fields.createListField('BOOLEAN', optionalFieldList);
-        case 'object':
-            return fields.createStructListField(fromJsonSchema(fieldValue.items), optionalFieldList);
-        default:
-            throw new UnsupportedJsonSchemaError(`Array field type ${JSON.stringify(fieldValue.items)} is unsupported.`);
-    }
+  switch (fieldValue.items.type) {
+    case "string":
+      if (fieldValue.items.format && fieldValue.items.format == "date-time") {
```

Comment:
Not a change at all, but I cannot wait for https://github.com/tc39/proposal-pattern-matching